File: src/nodejs/sample.js
var nodejsPort = require('./nodejs_port');

var testClassObject = nodejsPort.TestClass.RawFactoryFunction();

var randomInt = Math.round((Math.random() * (1000 - 1)) + 1);

console.log('Random Int:', randomInt);

console.log('Passing var to C++');
testClassObject.SetMyValue(randomInt);

console.log('Getting var from C++:', testClassObject.GetMyValue());

console.log('Getting Random UInt from MS-Common>utils.h:', testClassObject.GetRandomUInt32())

console.log('\nPassing 86.05 to the function templates below:')
console.log('Getting int from function template:', testClassObject.GetTemplateValueAsInt(86.05))
console.log('Getting char from function template:', testClassObject.GetTemplateValueAsChar(86.05))
console.log('Getting double from function template:', testClassObject.GetTemplateValueAsDouble(86.05))

console.log('\nTestClass::PrintMe():', testClassObject.PrintMe());
testClassObject.PrintLogMessage();

console.log('\nDone...');

File: src/python/sample.py
import python_port
import random

testClassObject = python_port.TestClass.RawFactoryFunction()
randomInt = random.randint(1, 1000)
print "{} {}".format("Random Int:", randomInt)
print "Passing var to C++"
testClassObject.SetMyValue(randomInt)
print "{} {}".format("Getting var from C++:", testClassObject.GetMyValue())
print "{} {}".format("Getting Random UInt from MS-Common>utils.h:", testClassObject.GetRandomUInt32())

print "\nPassing 86 to the int & char function templates and 86.05 to the double function template below:"
print "{} {}".format("Getting int from function template:", testClassObject.GetTemplateValueAsInt(86))
print "{} {}".format("Getting char from function template:", testClassObject.GetTemplateValueAsChar(86))
print "{} {}".format("Getting double from function template:", testClassObject.GetTemplateValueAsDouble(86.05))

print "{} {}".format("\nTestClass::PrintMe():", testClassObject.PrintMe())

print "\nEncoding - Some String"
hexEncodedString = testClassObject.HexEncode("Some String")
print "{} {}".format("Hex:", hexEncodedString)
print "{} {}".format("Decoded:", testClassObject.HexDecode(hexEncodedString))
print "\nDone..."
